src/averageAges.js

```diff
@@ -37,7 +37,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenFilterAges = people.filter(person =>
+    withChildren
+      ? people.find(child => child.mother === person.name) && person.sex === 'f'
+      : person.sex === 'f'
+  );
+
+  return womenFilterAges.reduce((sum, person) => sum
```

Comment:
This calculates is common to several functions - to avoid duplication, you can move it to a separate function 🙌